src/main/java/io/github/railroad/vcs/connections/hubs/GithubConnection.java

```diff
@@ -0,0 +1,218 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class GithubConnection extends AbstractConnection {
+    private Profile account;
+
+    public GithubConnection(Profile profile) {
+        this.account = profile;
+    }
+
+    private List<HttpResponse> readHTTP(String method, String postixurl, String body) throws Exception {
+        return readHTTP(method, postixurl, body, true);
+    }
+
+    private List<HttpResponse> readHTTP(String method, String postixurl, String body, boolean enable_pages) throws Exception {
+        if (account.getAccessToken().isEmpty()) {
+            throw new Exception("Missing Access Token");
+        }
+        List<HttpResponse> result = new ArrayList<>();
+
+        String request_url = "https://api.github.com/" + postixurl;
+        boolean finished = false;
+        try {
+            try {
+
+                while (!finished) {
+                    URL url = new URL(request_url);
+                    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+                    con.setRequestMethod(method.toUpperCase());
+                    con.setRequestProperty("Accept", "application/vnd.github+json");
+                    con.setRequestProperty("X-GitHub-Api-Version", "2022-11-28");
+                    con.setRequestProperty("Authorization", "Bearer " + account.getAccessToken());
+                    BufferedReader in = new BufferedReader(
+                            new InputStreamReader(con.getInputStream()));
+                    String inputLine;
+                    StringBuffer content = new StringBuffer();
+                    var statusCode = con.getResponseCode();
+                    while ((inputLine = in.readLine()) != null) {
+                        content.append(inputLine);
+                    }
+                    if (enable_pages) {
+                        if (con.getHeaderField("link") != null) {
+                            finished = true;
+                            for (String el : con.getHeaderField("link").split(",")) {
+                                if (el.contains("rel=\"next\"")) {
+                                    request_url = el.substring(el.indexOf("<") + 1, el.indexOf(">"));
+                                    finished = false;
+                                }
+                            }
+                        } else {
+                            finished = true;
+                        }
+                    } else {
+                        finished = true;
+                    }
+                    in.close();
+                    result.add(new HttpResponse(content.toString(), statusCode));
+                }
+
+            } catch (ProtocolException e) {
+                throw new RuntimeException(e);
+            } catch (MalformedURLException e) {
+                throw new RuntimeException(e);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
These can be collapsed

Reply:
what do you mean

Reply:
`ProtocolException | MalformedURLException | IOException exception`

Reply:
i think like this is better do you mind checking please?